sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVDataSource.scala

```diff
@@ -155,7 +155,7 @@ object TextInputCSVDataSource extends CSVDataSource {
         sparkSession,
         paths = paths,
         className = classOf[TextFileFormat].getName,
-        options = options.parameters
+        options = options.parameters - "path"
```

Comment:
This is called during `infer` and "path" option gets added to `inputPaths`, thus this needs to be removed.

Reply:
I also thought about update the following
https://github.com/apache/spark/blob/8659ec554fe1993fcaf2ec4d1c4745205bddec7d/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L575

to

```
val allPaths = caseInsensitiveOptions.get("path").map(Seq(_)).getOrElse(paths)
```
, but it looked like a risky change (e.g., there is no good place for assert, etc.).

Reply:
This is a good catch! BTW does it have to be fixed in this PR? The check is added in `DataFrameReader.load` and `DataSource.resolveRelation` should be fine.

Reply:
Yes, without this, few tests that involve "inferring" fail.

Basically, when you do `spark.read.csv("/tmp")`, "path" option is populated since there is only one path parameter in `load()`. Now, when `infer` is called, `inputPaths: Seq[FileStatus]` is a result of listing files and there is the "path" option that is being carried over, so `infer` will try to read more data than needed.

Reply:
Can you add some comments here at least? I feel it will confuse code readers later on why removing it...

Reply:
Thanks @viirya for pointing it out. This is now addressed in #29437.